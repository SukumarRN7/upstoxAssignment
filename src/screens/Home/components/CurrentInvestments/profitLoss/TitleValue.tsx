import React from 'react';
import {StyleProp, StyleSheet, Text, View, ViewStyle} from 'react-native';

interface ITitleValue {
  title: string;
  value: string;
  style?: StyleProp<ViewStyle>;
}

export const TitleValue: React.FC<ITitleValue> = ({title, value, style}) => {
  return (
    <View style={[styles.container, style]}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.value}>{value}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 20,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
  },
  title: {fontWeight: 'bold', fontSize: 16},
  value: {
    fontWeight: '600',
  },
});
