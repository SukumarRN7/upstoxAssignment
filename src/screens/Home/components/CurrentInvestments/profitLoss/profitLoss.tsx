import React from 'react';
import {StyleSheet, View} from 'react-native';
import {APP_THEME} from '../../../../../theme';
import Accordian from '../../../../../components/accordian';
import {TitleValue} from './TitleValue';
import {TEXT_SYMBOLS} from '../../../../../assets/textSymbols';
import {TProfitLossInfo} from '../../../../../hooks/currentInvestments/type';

interface IProfitLoss {
  info: TProfitLossInfo;
}

const infoSet = {
  'Current Value': 0,
  'Total Investment': 0,
  "Today's profit & Loss": 0,
};

export const ProfitLoss: React.FC<IProfitLoss> = ({info}) => {
  const {PNL, currentValueTotal, investedValueTotal, todayPNL} = info;

  const getValue = (idx: number) => {
    switch (idx) {
      case 0:
        return `${TEXT_SYMBOLS.INR} ${currentValueTotal.toFixed(2)}`;
      case 1:
        return `${TEXT_SYMBOLS.INR} ${investedValueTotal.toFixed(2)}`;
      case 2:
        return `${TEXT_SYMBOLS.INR} ${todayPNL.toFixed(2)}`;
      default:
        return '';
    }
  };

  return (
    <View style={styles.container}>
      <Accordian>
        <View>
          {Object.keys(infoSet).map((item, idx) => {
            return (
              <TitleValue
                key={idx}
                title={item}
                value={getValue(idx)}
                style={styles.infoSetTitleValue}
              />
            );
          })}
        </View>
      </Accordian>
      <View style={styles.profitLossContainer}>
        <TitleValue
          title={'Profit & Loss'}
          value={`${TEXT_SYMBOLS.INR} ${PNL.toFixed(2)}`}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 0,
    width: '100%',
    backgroundColor: APP_THEME.colors.white,
    paddingHorizontal: 20,
    paddingTop: 10,
  },
  profitLossContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 20,
  },
  infoSetTitleValue: {
    paddingBottom: 2,
  },
});
