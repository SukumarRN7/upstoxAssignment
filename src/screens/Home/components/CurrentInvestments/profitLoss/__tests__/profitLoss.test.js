import React from 'react';
import renderer from 'react-test-renderer';
import {ProfitLoss} from '../profitLoss';

describe('ProfitLoss', () => {
  const infoData = {
    PNL: 1000,
    currentValueTotal: 2000,
    investedValueTotal: 1500,
    todayPNL: 500,
  };

  test('renders correctly', () => {
    const component = renderer.create(<ProfitLoss info={infoData} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
