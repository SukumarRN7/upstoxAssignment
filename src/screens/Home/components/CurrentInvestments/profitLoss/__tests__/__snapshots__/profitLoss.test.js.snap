// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfitLoss renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "bottom": 0,
      "paddingHorizontal": 20,
      "paddingTop": 10,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        collapsable={false}
        jestAnimatedStyle={
          {
            "value": {
              "alignSelf": "center",
              "transform": [
                {
                  "rotate": "0deg",
                },
              ],
            },
          }
        }
        style={
          {
            "alignSelf": "center",
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderBottomColor": "purple",
                "borderBottomWidth": 20,
                "borderLeftColor": "transparent",
                "borderLeftWidth": 10,
                "borderRightColor": "transparent",
                "borderRightWidth": 10,
                "borderStyle": "solid",
                "height": 0,
                "width": 0,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "height": 0,
            "opacity": 0,
          },
        }
      }
      style={
        {
          "height": 0,
          "opacity": 0,
        }
      }
    >
      <View
        collapsable={false}
        forwardedRef={[Function]}
        jestAnimatedStyle={
          {
            "value": {},
          }
        }
        style={
          {
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <View>
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 20,
                  "width": "100%",
                },
                {
                  "paddingBottom": 2,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Current Value
            </Text>
            <Text
              style={
                {
                  "fontWeight": "600",
                }
              }
            >
              ₹ 2000.00
            </Text>
          </View>
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 20,
                  "width": "100%",
                },
                {
                  "paddingBottom": 2,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Total Investment
            </Text>
            <Text
              style={
                {
                  "fontWeight": "600",
                }
              }
            >
              ₹ 1500.00
            </Text>
          </View>
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 20,
                  "width": "100%",
                },
                {
                  "paddingBottom": 2,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Today's profit & Loss
            </Text>
            <Text
              style={
                {
                  "fontWeight": "600",
                }
              }
            >
              ₹ 500.00
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 20,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Profit & Loss
      </Text>
      <Text
        style={
          {
            "fontWeight": "600",
          }
        }
      >
        ₹ 1000.00
      </Text>
    </View>
  </View>
</View>
`;
